backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,342 @@
+                                // check priority of attributes
+                                if (gatheredAttributes[a.Name].Priority < a.Priority)
+                                {
+                                    gatheredAttributes[a.Name] = a;
+                                }
+                            }
+                            else
+                            {
+                                gatheredAttributes.Add(a.Name, a);
+                            }
+                        }
+                    }
+                }
+
+
+                foreach (var followRelation in cfg.BaseCI.FollowRelations)
+                {
+                    var prevCI = mainCI;
+                    foreach (var follow in followRelation.Follow)
+                    {
+                        // get the predicate remove the first char, first char defines the direction of relation
+                        var predicate = follow.Predicate[1..];
+
+                        // check relation direction
+                        var r = allRelations.Where(r =>
+                        {
+                            if (follow.Predicate[0] == '>')
+                            {
+                                return r.FromCIID == prevCI.ID && r.PredicateID == predicate;
+                            }
+                            else
+                            {
+                                return r.ToCIID == prevCI.ID && r.PredicateID == predicate;
+                            }
+                        }).FirstOrDefault();
+
+
+                        if (r == null)
+                        {
+                            break;
+                        }
+
+                        var targetCI = allCIs.Where(ci =>
+                        {
+                            if (follow.Predicate[0] == '>')
+                            {
+                                return r.ToCIID == ci.ID;
+                            }
+                            else
+                            {
+                                return r.FromCIID == ci.ID;
+                            }
+                        }
+                        ).FirstOrDefault();
+
+                        prevCI = targetCI;
+
+                        var targetCIRequiredTrait = await traitsProvider.GetActiveTrait(follow.RequiredTrait, trans, changesetProxy.TimeThreshold);
+
+                        if (targetCIRequiredTrait == null)
+                        {
+                            logger.LogError($"The required trait {follow.RequiredTrait} for follow realtions was not found!");
+                            continue;
+                        }
+
+                        // check if target CI has the required trait
+                        var aa = await effectiveTraitModel.FilterCIsWithTrait(new List<MergedCI> { targetCI }, targetCIRequiredTrait, layersetVariableRendering, trans, changesetProxy.TimeThreshold);
+
+                        if (aa.ToList().Count == 0)
```

Comment:
you don't always need to call ToList(), IEnumerable<> has a .Count() method you can use here

Reply:
I noticed that here Any() can be used for more efficiency.